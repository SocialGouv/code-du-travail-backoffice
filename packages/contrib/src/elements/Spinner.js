import React from "react";
import styled, { keyframes } from "styled-components";

const Container = styled.div`
  display: inline-block;
  position: relative;
  width: ${props => props.size};
  height: ${props => props.size};
`;

const ripple = keyframes`
  0% {
    top: 28px;
    left: 28px;
    width: 0;
    height: 0;
    opacity: 1;
  }
  100% {
    top: -1px;
    left: -1px;
    width: 58px;
    height: 58px;
    opacity: 0;
  }
`;
const Spinner = styled.div`
  position: absolute;
  border: 4px solid #666666;
  opacity: 1;
  border-radius: 50%;
  animation: ${ripple} 1s cubic-bezier(0, 0.2, 0.8, 1) infinite;
`;
const SpinnerNext = styled(Spinner)`
  animation-delay: -0.5s;
`;

export default ({ size }) => (
  <Container size={size !== undefined ? size : "4rem"}>
    <Spinner />
    <SpinnerNext />
  </Container>
);
