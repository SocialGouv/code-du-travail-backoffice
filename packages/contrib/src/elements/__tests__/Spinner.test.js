import React from "react";
import { render } from "react-testing-library";
import Spinner from "../Spinner";

describe("[Contrib] elements/<Spinner />", () => {
  it("should match snapshot", () => {
    const { container } = render(<Spinner size="8rem" />);

    expect(container).toMatchSnapshot();
  });
});
