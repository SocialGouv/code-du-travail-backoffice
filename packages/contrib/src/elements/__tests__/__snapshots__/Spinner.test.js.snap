// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Contrib] elements/<Spinner /> should match snapshot 1`] = `
<div>
  <div
    class="Spinner__Container-nm5wf5-0 iybyio"
  >
    <div
      class="Spinner-nm5wf5-1 cCjJqj"
    />
    <div
      class="Spinner-nm5wf5-1 Spinner__SpinnerNext-nm5wf5-2 hUyNhV"
    />
  </div>
</div>
`;
